pilot/pkg/model/telemetry.go

```diff
@@ -1052,3 +1055,51 @@ func generateStatsConfig(class networking.ListenerClass, metricsCfg telemetryFil
 func disableHostHeaderFallback(class networking.ListenerClass) bool {
 	return class == networking.ListenerClassSidecarInbound || class == networking.ListenerClassGateway
 }
+
+type OTelResourceSemantic string
+
+const (
+	KubernetesCluster   OTelResourceSemantic = "k8s.cluster.name"
+	KubernetesNamespace OTelResourceSemantic = "k8s.namespace.name"
+	KubernetesPod       OTelResourceSemantic = "k8s.pod.name"
+	ServiceName         OTelResourceSemantic = "service.name"
+)
+
+// resourceAttributes return semantic attributes follow OpenTelemetry specification
+// see more details here: https://opentelemetry.io/docs/reference/specification/resource/semantic_conventions/
+func resourceAttributes(proxy *Proxy) *otlpcommon.KeyValueList {
+	// TODO(zirain): generate resource attributes for VM
+	if proxy.IsVM() {
+		return nil
+	}
+
+	podName, _ := extractProxyID(proxy)
+
+	return &otlpcommon.KeyValueList{
+		Values: []*otlpcommon.KeyValue{
+			otelKeyValue(otelsemconv.K8SClusterNameKey, proxy.Metadata.ClusterID.String()),
+			otelKeyValue(otelsemconv.K8SNamespaceNameKey, proxy.ConfigNamespace),
+			otelKeyValue(otelsemconv.K8SPodNameKey, podName),
+			// this seems a little hack, but in istio it works, we do the samething for `serviceCluster` in tracing
+			otelKeyValue(otelsemconv.ServiceNameKey, proxy.XdsNode.Cluster),
+		},
+	}
+}
+
+func otelKeyValue(semantic attribute.Key, val string) *otlpcommon.KeyValue {
+	return &otlpcommon.KeyValue{
+		Key:   string(semantic),
+		Value: &otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_StringValue{StringValue: val}},
+	}
+}
+
+func extractProxyID(proxy *Proxy) (string, string) {
```

Comment:
Can we please use the same logic in baggage and here?

Reply:
refactor with pkg go.opentelemetry.io/otel/baggage, PTAL again.

Reply:
I meant the business logic of populating the fields more so than the `go.opentelemetry.io/otel/baggage` package.. although that is interesting, I didn't know they had a package. It seems like just a very inefficient version of fmt.Sprintf though....


Its not clear to me we should be using `k8s.pod.name` in all cases. In other telemetry aspects, we are using WorkloadName, which is quite different - even on k8s, this is not always a Pod, but more often the Deployment name. There is currently a `TODO do not hardcode deployment. But I think we ignore it anyways?` in the Go code, but in the proxy code (`istio.telemetry.baggagehandler.v1.Config`). The proxy already actually returns the info we need to access this via OWNER.

More concretely, we shouldn't be parsing ID here, beyond the initial computation of model.Proxy it should be thought of as an opaque string. There are other, more reliable, ways to access the info we are after.